sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -221,17 +221,20 @@ def plot(self, plot_method="contourf", ax=None, xlabel=None, ylabel=None, **kwar
             self.surface_ = plot_func(self.xx0, self.xx1, self.response, **kwargs)
         else:  # self.response.ndim == 3
             n_responses = self.response.shape[-1]
-            if (
-                isinstance(self.multiclass_colors, str)
-                or self.multiclass_colors is None
+            if "cmap" in kwargs:
+                warnings.warn("'cmap' is ignored in multiclass case.")
+                del kwargs["cmap"]
+            if "colors" in kwargs:
+                warnings.warn("'colors' is ignored in multiclass case.")
+                del kwargs["colors"]
```

Comment:
Small nitpick: let's keep the mention of `decision_function` and `predict_proba` in the warning message, because it doesn't apply to `predict`. Something like 
```
'cmap' is ignored in favor of 'multiclass_colors' in the multiclass case when the response method is 'decision_function' or 'predict_proba'.
```